index.js

```diff
@@ -39,73 +39,103 @@ const getPackageJson = function() {
   return pkg;
 };
 
-const defaultGlobals = {
-  browser: {
-    'video.js': 'videojs',
-    'global': 'window',
-    'global/window': 'window',
-    'global/document': 'document'
-  },
-  module: {
-    'video.js': 'videojs'
-  },
-  test: {
-    'qunit': 'QUnit',
-    'qunitjs': 'QUnit',
-    'sinon': 'sinon',
-    'video.js': 'videojs'
-  }
-};
-
-const defaultExternals = {
-  browser: [],
-  module: [
-    'global',
-    'global/document',
-    'global/window'
-  ],
-  test: []
-};
-
-const defaultPlugins = {
-  browser: [
-    'resolve',
-    'json',
-    'commonjs',
-    'uglify',
-    'babel'
-  ],
-
-  module: [
-    'resolve',
-    'json',
-    'commonjs',
-    'babel'
-  ],
+// if checkWatch is true and -w or --watch was passed to rollup
+// we should "change the build" for rollups watch mode.
+const shouldChangeWatch = (settings) =>
+  settings.checkWatch && process.argv.some((arg) => (/^-w|--watch$/).test(arg));
+
+// verify that listed plugins are an object or in primed plugins
+const validatePlugins = ({plugins, primedPlugins}) =>
+  Object.keys(plugins).forEach((buildType) => plugins[buildType].forEach((pluginName) => {
+    if (typeof pluginName !== 'string') {
+      return;
+    }
 
-  test: [
-    'multiEntry',
-    'resolve',
-    'json',
-    'commonjs',
-    'istanbul',
-    'babel'
-  ]
+    if (!primedPlugins[pluginName]) {
+      throw new Error(`Plugin ${pluginName} listed in ${buildType} does not have an equivent Primed Plugin`);
+    }
+  }));
+
+// all complex defaults
+// Note: the order here matters because one default often
+// determines the value of another default
+const ORDERED_DEFAULTS = {
+  globals: () => ({
```

Comment:
These are now functions because: 
1. Most of these "defaults" need the build settings to be set correctly
2. We want a new object each run so that `rollup --watch` does not continuously change objects.